src/public/components/router.js

```diff
@@ -1,6 +1,83 @@
-import K, * as U from "karet.util"
-import React     from "karet"
+import * as R     from "ramda"
+import * as L     from "partial.lenses"
+import K, * as U  from "karet.util"
+import React      from "karet"
+import toRegex    from 'path-to-regexp'
 
-export default U.withContext(({routes, NotFound}, {path}) => U.fromKefir(K(path, path =>
-  React.createElement(routes[path] || NotFound)
-)))
+const W = a => b => a( b )( b )
+
+// TODO: Make it possible to add routes more than once?
+
+// begin routes initialization
+const partitionDynamicAndStatic =
+  R.partition( R.compose( R.lt( 0 )
+                        , R.length
+                        , L.get( [ 'regex', 'keys' ] )
+                        )
+             )
+
+const addRegexes =
+  R.map( W( R.compose( L.set( 'regex' )
+                     , toRegex
+                     , L.get( 'route' )
+                     )
+          )
+       )
+
+const expandRoutes =
+  R.compose( R.map( R.zipObj( [ 'route', 'component' ] ) )
+           , R.toPairs
+           )
+
+const init =
+  R.compose( R.zipObj( [ 'dynamic', 'static' ] )
+           , partitionDynamicAndStatic
+           , addRegexes
+           , expandRoutes
+           )
+// end routes initialization
+
+//
+
+// begin route resolution
+const resolveStatic = path =>
+  R.compose( R.find( R.propEq( 'route', path ) )
+           , L.get( 'static' )
+           )
+
+const prepParams =
+  R.compose( R.zipObj
+           , R.map( L.get( 'name' ) )
+           )
+
+const resolveDynamic = path =>
+  R.compose( R.reduce( ( acc, { component, route, regex } ) =>
+                       { const match = regex.exec( path )
+                         return match !== null
+                                ? R.reduced( { component, path, route, regex
+                                             , params: prepParams( regex.keys )( R.tail( match ) )
```

Comment:
I had actually already made this change on my side, last night (although I hadn't renamed `params` to `props`. I vacillated between passing everything into the component and just passing the props/params, but figured we'd settle on just passing the props and dropping the rest.

I'll rename `params` to `props`.